salt/modules/elasticsearch.py

```diff
@@ -373,7 +389,13 @@ def document_create(index, doc_type, body=None, id=None, hosts=None, profile=Non
         salt myminion elasticsearch.document_create testindex doctype1 '{}'
     '''
     es = _get_instance(hosts, profile)
-
+    if source and body:
+        message = 'Either body or source should be specified but not both.'
```

Comment:
IMHO, we need to document these sorts of limitation in the function documentation.